classes/user/DAO.inc.php

```diff
@@ -0,0 +1,344 @@
+        'disabled' => 'disabled',
+        'disabledReason' => 'disabled_reason',
+        'inlineHelp' => 'inline_help',
+    ];
+
+    /* These constants are used user-selectable search fields. */
+    public const USER_FIELD_USERID = 'user_id';
+    public const USER_FIELD_USERNAME = 'username';
+    public const USER_FIELD_EMAIL = 'email';
+    public const USER_FIELD_URL = 'url';
+    public const USER_FIELD_INTERESTS = 'interests';
+    public const USER_FIELD_AFFILIATION = 'affiliation';
+    public const USER_FIELD_NONE = null;
+
+    /**
+     * Construct a new User object.
+     *
+     * @return User
+     */
+    public function newDataObject()
+    {
+        return new User();
+    }
+
+    /**
+     * @copydoc EntityDAO::get()
+     *
+     * @param $allowDisabled boolean If true, allow fetching a disabled user.
+     */
+    public function get($id, $allowDisabled = true): ?User
+    {
+        $user = parent::get($id);
+        if (!$allowDisabled && $user->getDisabled()) {
+            return null;
+        }
+        return $user;
+    }
+
+    /**
+     * Get a collection of announcements matching the configured query
+     */
+    public function getMany(Collector $query): LazyCollection
+    {
+        $rows = $query
+            ->getQueryBuilder()
+            ->get();
+
+        return LazyCollection::make(function () use ($rows, $query) {
+            foreach ($rows as $row) {
+                yield $row->user_id => $this->fromRow($row, $query->includeReviewerData);
+            }
+        });
+    }
+
+    /**
+     * Get the total count of users matching the configured query
+     */
+    public function getCount(Collector $query): int
+    {
+        return $query
+            ->getQueryBuilder()
+            ->count();
+    }
+
+    /**
+     * Get a list of ids matching the configured query
+     */
+    public function getIds(Collector $query): Collection
+    {
+        return $query
+            ->getQueryBuilder()
+            ->select('u.' . $this->primaryKeyColumn)
+            ->pluck('u.' . $this->primaryKeyColumn);
+    }
+
+    /**
+     * Retrieve a user by username.
+     *
+     * @param $username string
+     * @param $allowDisabled boolean
+     *
+     * @return User?
+     */
+    public function getByUsername(string $username, bool $allowDisabled = true): ?User
+    {
+        $row = DB::table('users')
+            ->where('username', '=', $username)
+            ->when(!$allowDisabled, function ($query) {
+                return $query->where('disabled', '=', false);
+            })
+            ->get('user_id')
+            ->first();
+        return $row ? $this->get($row->user_id) : null;
+    }
+
+    /**
+     * Retrieve a user by API key.
+     *
+     * @param $allowDisabled boolean
+     *
+     * @return User?
+     */
+    public function getByApiKey(string $apiKey, bool $allowDisabled = true): ?User
+    {
+        return $this->getBySetting('apiKey', $apiKey, $allowDisabled);
+    }
```

Comment:
Seems to me like if we're going to add an `$allowedDisable` argument to the method it should be `false` by default.

Reply:
(See: https://github.com/pkp/pkp-lib/pull/7170/files#r692547579)